src/AI/AICombatUtils.java

```diff
@@ -207,13 +267,14 @@ public static int findMaxStrikeWeaponRange(Commander co)
 
   /**
    * Attempts to find a combination of attacks that will create a kill.
+   * Considers units in the order provided.
    * Recursive.
    * @param unitQueue The set of potential attackers
    * @param neededAttacks The set of locations to consider, pre-populated with any mandatory attacks, to be populated
    * @param pDamage The cumulative base damage done by those mandatory attacks
    * @return The cumulative base damage of all attacks in the neededAttacks
    */
-  public static double findAssaultKill(GameMap gameMap, Collection<Unit> unitQueue, Map<XYCoord, Unit> neededAttacks, Unit target, double pDamage)
+  public static double findMultiHitKill(GameMap gameMap, Collection<Unit> unitQueue, Map<XYCoord, Unit> neededAttacks, Unit target, double pDamage)
```

Comment:
The two functions are both public, but one considers units in cost order and the other in prior order? They are also a bit cavalier in terms of parameter ordering or naming. One is
GameMap, target, Commander, attackCandidates, excludedSpaces; the other is
GameMap, unitQueue, neededAttacks, target, pDamage

Reply:
> The two functions are both public, but one considers units in cost order and the other in prior order?

Right. The first one is basically a more user-friendly interface to the other function.
I'm not opposed to changing assumptions and such (let the caller pick the ordering or whatever).
The second function is public because it allows you to do stuff like partial calcs: "I know I want to use this unit to hit that one, but can I finish it off this turn?"

Your point about naming/order is fair, though.